### PR TITLE
Update RPC reference

### PR DESCRIPTION
**Pull request by alexandratran**
Update wallet JSON-RPC API reference for conciseness and clarity, including:

- Make link to the API playground more prominent.
- Update information about restricted methods - technically, only `eth_accounts` should be categorized as restricted.
- Integrate changes from #89.

Eventually, most of the RPC docs should be solely maintained on [`api-specs`](https://github.com/MetaMask/api-specs), and we can remove the documentation here.

### CODE REVIEW

**Comment by alexandratran** [comment]
@metamaskbot publish-preview

**Review by shanejonas — APPROVED** [review]

**Comment by alexandratran** [comment]
@metamaskbot publish-preview

File: wallet/reference/rpc-api.md [hunk 1]
```diff
@@ -1,95 +1,80 @@
...
-
-  // The date the permission was granted, in UNIX epoch time
-  date?: number;
-}
-```
+MetaMask introduced web3 wallet permissions in [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255).
+In this permissions system, each RPC method is restricted or unrestricted.
+If a method is restricted, the caller must request permission to call it using
+[`wallet_requestPermssions`](#walletrequestpermissions).
+Under the hood, permissions are plain, JSON-compatible objects, with fields that are mostly used
```
Comment (bgravenorst) [inline wallet/reference/rpc-api.md:40]: Not sure this sentence adds any value. Can probably remove?

File: wallet/reference/rpc-api.md [hunk 2]
```diff
@@ -1,95 +1,80 @@
...
-:::tip EIP-1102
 This method is specified by [EIP-1102](https://eips.ethereum.org/EIPS/eip-1102).
-It's equivalent to the deprecated [`ethereum.enable()`](provider-api.md#ethereumenable-deprecated)
-provider API method.
 
-Under the hood, it calls [`wallet_requestPermissions`](#wallet_requestpermissions) for the
-`eth_accounts` permission.
-Since `eth_accounts` is currently the only permission, this method is all you need for now.
+:::info
+Under the hood, this method calls [`wallet_requestPermissions`](#walletrequestpermissions) for
```
Comment (bgravenorst) [inline wallet/reference/rpc-api.md:69]:
```suggestion
Internally, this method calls [`wallet_requestPermissions`](#walletrequestpermissions) for
```

File: wallet/reference/rpc-api.md [hunk 3]
```diff
@@ -1,95 +1,80 @@
...
-}
-```
+MetaMask introduced web3 wallet permissions in [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255).
+In this permissions system, each RPC method is restricted or unrestricted.
+If a method is restricted, the caller must request permission to call it using
+[`wallet_requestPermssions`](#walletrequestpermissions).
+Under the hood, permissions are plain, JSON-compatible objects, with fields that are mostly used
+internally by MetaMask.
+
+Unrestricted methods have no corresponding permission, but they might still rely on permissions to
```
Comment (bgravenorst) [inline wallet/reference/rpc-api.md:43]: Should this info go in the unrestricted section?

**Review by bgravenorst — APPROVED** [review]
Added some feedback.

**Pull request merged**